### PR TITLE
Add some more `ruff` rules

### PR DESCRIPTION
**Pull request by harshil21**
Adds:

1. https://beta.ruff.rs/docs/rules/#flake8-async-async
2. https://beta.ruff.rs/docs/rules/#flake8-type-checking-tch
3. https://beta.ruff.rs/docs/rules/#perflint-perf
4. https://beta.ruff.rs/docs/rules/#copyright-related-rules-cpy
5. https://beta.ruff.rs/docs/rules/#flake8-slots-slot

### CODE REVIEW

File: telegram/ext/_aioratelimiter.py [hunk 1]
```diff
@@ -196,12 +196,11 @@ async def _run_request(
             self._get_group_limiter(group) if group and self._group_max_rate else null_context()
         )
 
-        async with group_context:  # skipcq: PTC-W0062
-            async with base_context:
-                # In case a retry_after was hit, we wait with processing the request
-                await self._retry_after_event.wait()
+        async with group_context, base_context:
```
Comment (Bibo-Joshi) [inline telegram/ext/_aioratelimiter.py:199]: does this guarantee that `group_context` is entered before the `base_context`?

File: tests/ext/test_jobqueue.py [hunk 2]
```diff
@@ -492,12 +492,10 @@ async def test_equality(self, job_queue):
         job_2 = job_queue.run_repeating(self.job_run_once, 0.2)
         job_3 = Job(self.job_run_once, 0.2)
         job_3._job = job.job
-        assert job == job
```
Comment (Bibo-Joshi) [inline tests/ext/test_jobqueue.py:495]: why? I'm rather sure that we do want to assert that … same below

**Review by Bibo-Joshi — COMMENTED** [review]
Meta question: I guess you kept track of the changelog and added the new rules manually?

File: telegram/ext/_aioratelimiter.py (hunk 1 above)
Comment (harshil21) [reply telegram/ext/_aioratelimiter.py:199]: yes, scroll down a bit on https://docs.python.org/3/reference/compound_stmts.html#with

File: tests/ext/test_jobqueue.py (hunk 2 above)
Comment (harshil21) [reply tests/ext/test_jobqueue.py:495]: ```PLR0124 Name compared with itself, consider replacing `job == job` ```

should I just ignore it then?

**Comment by harshil21** [comment]
> Meta question: I guess you kept track of the changelog and added the new rules manually?

at first no, I went through the whole list again, but with the newer versions released since this PR, yes I tracked the changelog

File: tests/ext/test_jobqueue.py (hunk 2 above)
Comment (Bibo-Joshi) [reply tests/ext/test_jobqueue.py:495]: yes, please

**Comment by Bibo-Joshi** [comment]
> > Meta question: I guess you kept track of the changelog and added the new rules manually?
> 
> at first no, I went through the whole list again, but with the newer versions released since this PR, yes I tracked the changelog

Okay. I'm wondering if we can automate that a bit … or at least do that manually on dependabot PRs …

**Comment by harshil21** [comment]
> Okay. I'm wondering if we can automate that a bit … or at least do that manually on dependabot PRs …

I'm okay with checking manually on dependabot PRs for now

**Review by Bibo-Joshi — APPROVED** [review]
LGTM :) let's wait for the master branch to work again before merging, though …

**Pull request merged**